File: pickerview/src/main/java/org/jaaksi/pickerview/picker/option/IOptionDelegate.java
package org.jaaksi.pickerview.picker.option;

import java.util.List;
import org.jaaksi.pickerview.dataset.OptionDataSet;
import org.jaaksi.pickerview.picker.OptionPicker;

/**
 * Created by fuchaoyang on 2018/7/6.<br/>
 * description：
 */

public interface IOptionDelegate {
  //void init(int hierarchy, List<PickerView> pickerViews, int[] selectedPosition);
  void init(OptionPicker.Delegate delegate);

  void setData(List<? extends OptionDataSet>... options);

  void setSelectedWithValues(String... values);

  OptionDataSet[] getSelectedOptions();

  void reset();
}

File: pickerview/src/main/java/org/jaaksi/pickerview/picker/option/IOptionDelegate.kt
package org.jaaksi.pickerview.picker.option

import org.jaaksi.pickerview.dataset.OptionDataSet
import org.jaaksi.pickerview.picker.OptionPicker

/**
 * Created by fuchaoyang on 2018/7/6.<br></br>
 * description：
 */
interface IOptionDelegate {
    fun init(delegate: OptionPicker.Delegate)

    fun setData(vararg options: List<OptionDataSet>)

    /**
     * 根据选中的values初始化选中的position
     */
    fun setSelectedWithValues(vararg values: String?)

    /**
     * 获取选中的选项，如果指定index为null则表示该列没有数据
     */
    val selectedOptions: Array<OptionDataSet?>
    fun reset()
}